controllers/component_build_controller.go

```diff
@@ -190,10 +190,44 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		// Also it can contain github-private-key and github-application-id
 		// in case GitHub Application is used instead of webhook.
 		pacSecret := corev1.Secret{}
-		if err := r.Client.Get(ctx, types.NamespacedName{Namespace: pipelinesAsCodeNamespace, Name: gitopsprepare.PipelinesAsCodeSecretName}, &pacSecret); err != nil {
-			log.Error(err, "failed to get Pipelines as Code secret")
-			r.EventRecorder.Event(&pacSecret, "Warning", "ErrorReadingPaCSecret", err.Error())
-			return ctrl.Result{}, err
+		localPaCSecretKey := types.NamespacedName{Namespace: component.Namespace, Name: gitopsprepare.PipelinesAsCodeSecretName}
+		if err := r.Client.Get(ctx, localPaCSecretKey, &pacSecret); err != nil {
+			if !errors.IsNotFound(err) {
+				log.Error(err, fmt.Sprintf("failed to get Pipelines as Code secret in %s namespace", component.Namespace))
+				r.EventRecorder.Event(&pacSecret, "Warning", "ErrorReadingPaCSecret", err.Error())
+				return ctrl.Result{}, err
+			}
+
+			// Fallback to the global configuration
+			if err := r.LocalClient.Get(ctx, types.NamespacedName{Namespace: buildServiceNamespaceName, Name: gitopsprepare.PipelinesAsCodeSecretName}, &pacSecret); err != nil {
```

Comment:
would it be possible to skip the `Namespace` so it would be created in namespace where the operator is running?

Reply:
No, the `namespace` field is mandatory because it's part of object identifier.